### PR TITLE
Allows HTTP timeout to be configured.

### PR DESCRIPTION
**Pull request by djvirgen**
// In node:
    app.use(TaggedApi.middleware(host, {
      timeout: 30000 // 30 seconds
    });

    // In Angular
    app.config(function(taggedApiProvider) {
      taggedApiProvider.timeout = 30000; // 30 seconds
    });

### CODE REVIEW

File: lib/http_adapter/angular.js [hunk 1]
```diff
@@ -8,8 +8,13 @@
     function AngularAdapter($http, $window) {
         this._$http = $http;
         this._$window = $window;
+        this._timeout = 10000;
     }
 
+    AngularAdapter.prototype.setTimeout = function(timeout) {
```
Comment (skylargivens) [inline lib/http_adapter/angular.js]: It seems odd to use `setTimeout` as a method name when it's also a built-in JS function... I can't think of a better one though.

**Comment by eviechan** [comment]
LGTM :ship: :cherries:

File: lib/http_adapter/angular.js (hunk 1 above)
Comment (djvirgen) [inline lib/http_adapter/angular.js]: I agree, I could probably generalize this a bit more to allow for arbitrary HTTP options. Something like:

    AngularAdapter.prototype.setHttpOptions({
        timeout: 30000
    });

File: lib/index.js [hunk 2]
```diff
@@ -58,6 +61,10 @@
             }.bind(this));
         }
 
+        var timeout = parseInt(this._options.timeout, 10) || 10000;
+        if (timeout < 0) timeout = 10000;
```
Comment (skylargivens) [inline lib/index.js]: Do you think we'll ever want to change this from 10s? Perhaps it would be good to use a constant for this value?

**Comment by skylargivens** [comment]
:dancer: LGTM :+1:

File: lib/http_adapter/angular.js (hunk 1 above)
Comment (skylargivens) [inline lib/http_adapter/angular.js]: Good idea. If there are any extra options we'll add in the future, this seems like a good path.

File: lib/index.js (hunk 2 above)
Comment (djvirgen) [inline lib/index.js]: 10s seems like a good default, but due to international mobile users i want to increase to 30s in daiquiri.

Comment (djvirgen) [inline lib/index.js]: A constant is a good idea.

**Comment by djvirgen** [comment]
Merging for now to get this out, will have to clean up later.

**Pull request merged**